src/lib/transform.ts

```diff
@@ -0,0 +1,186 @@
+import type { JSONSchema, JSONSchemaDefinition } from './jsonschema';
+
+export function toStrictJsonSchema(schema: JSONSchema): JSONSchema {
+  return ensureStrictJsonSchema(schema, [], schema);
+}
+
+function isNullable(schema: JSONSchemaDefinition): boolean {
+  if (typeof schema === 'boolean') {
+    return false;
+  }
+  for (const oneOfVariant of schema.oneOf ?? []) {
+    if (typeof oneOfVariant !== 'boolean' && oneOfVariant.type === 'null') {
+      return true;
+    }
```

Comment:
this should be recursive

```suggestion
    if (isNullable(oneOfVariant)) {
      return true;
    }
```
also looks like `isNullable({ type: 'null' })` isn't handled

Reply:
fixed, thanks!